src/main/java/hudson/plugins/git/extensions/GitSCMExtension.java

```diff
@@ -161,6 +161,21 @@ public Revision decorateRevisionToBuild(GitSCM scm, AbstractBuild<?,?> build, Gi
      * @throws GitException on git error
      */
     public void beforeCheckout(GitSCM scm, Run<?,?> build, GitClient git, TaskListener listener) throws IOException, InterruptedException, GitException {
+        boolean isPipeline =  build != null && build.getClass().getName().startsWith("org.jenkinsci.plugins.workflow.job.");
+
+        if (isPipeline) {
+
+            // Case 1: MessageExclusion (special-case, higher priority)
+            if (this.getClass().getName().equals("hudson.plugins.git.extensions.impl.MessageExclusion")) {
+                listener.getLogger().println("DEPRECATED: Message exclusion is deprecated for Pipeline jobs. " + "Use the Pipeline SCM trait instead.");
+
+                // Case 2: Polling-related deprecated extensions
+            } else if (requiresWorkspaceForPolling()) {
+
+                listener.getLogger().println("DEPRECATED: This Git extension is deprecated for Pipeline jobs and forces workspace polling. " + "Use Pipeline-native SCM polling instead.");
```

Comment:
I'd prefer that the message specifically describe the extension that is deprecated instead of using "This extention"/
```suggestion
                listener.getLogger().println("DEPRECATED: The extension that requires a workspace for polling is deprecated for Pipeline jobs. " + "Use Pipeline-native SCM polling instead.");
```